src/main/java/org/opensearch/knn/index/KNNSettings.java

```diff
@@ -289,6 +292,9 @@ public class KNNSettings {
         }
     };
 
+    private static Map<String, Setting<?>> FEATURE_FLAGS = KNNLaunchControl.getFeatureFlags()
```

Comment:
Can we use  Collections.unmodifiableMap to ensure that the map is immutable, which is a good practice for feature flags. Additionally can we mark this as final , just to prevent this getting re-assigned.